tree_hash_derive/src/lib.rs

```diff
@@ -224,11 +502,11 @@ fn tree_hash_derive_enum_transparent(
                 #(
                     assert_eq!(
```

Comment:
This assertion seems to prevent you from using superstruct with a transparent (or transparent_stable) `tree_hash` marker where one variant is a normal container and the other is a profile. This seems like a useful thing to do. Is there a way around this?